src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1702,10 +1664,52 @@ private void SetNotNullResult(BoundExpression node)
             SetResultType(node, TypeWithState.Create(node.Type, NullableFlowState.NotNull));
         }
 
+        /// <summary>
+        /// Returns true if the type is a struct with no fields or properties.
+        /// </summary>
+        protected override bool IsEmptyStructType(TypeSymbol type)
+        {
+            if (type.TypeKind != TypeKind.Struct)
+            {
+                return false;
+            }
+
+            // EmptyStructTypeCache.IsEmptyStructType() returns false
+            // if there are non-cyclic fields.
+            if (!_emptyStructTypeCache.IsEmptyStructType(type))
+            {
+                return false;
+            }
+
+            if (type.SpecialType != SpecialType.None)
+            {
+                return true;
+            }
+
+            var members = ((NamedTypeSymbol)type).GetMembersUnordered();
+
+            // EmptyStructTypeCache.IsEmptyStructType() returned true. If there
+            // are fields, those must be cyclic, so treat the type as empty.
```

Comment:
>those must be cyclic [](start = 27, length = 20)

Is this accurate? Are all of them cyclic?

Reply:
No it's not accurate. At least one field is cyclic. I'll update in a subsequent PR.

---
In reply to: [285217976](https://github.com/dotnet/roslyn/pull/35765#discussion_r285217976) [](ancestors = 285217976)